backend/inspirehep/records/marshmallow/literature/cds.py

```diff
@@ -54,12 +62,21 @@ class LiteraturePublicSchemaCDS(LiteraturePublicSchema):
     @staticmethod
     def get_authors(data):
         updated_authors = []
+        author_records = InspireRecord.get_linked_records_from_dict_field(
+            data, "authors.record"
+        )
+        institution_records = InspireRecord.get_linked_records_from_dict_field(
+            data, "authors.affiliations.record"
+        )
+        resolved_authors_by_id = {
+            record["control_number"]: record for record in author_records
+        }
+        resolved_institutions_by_id = {
+            str(record["control_number"]): record for record in institution_records
```

Comment:
why we use here `str` and not for `authors`?

Reply:
Becuase for the institutions one later in the function we do 

https://github.com/inspirehep/inspirehep/pull/3604/files#diff-f616257fa08231f84a2654e38bef46c7648caae30e21598a86c13f8f9f49b361R37-R38

This returns a tuple of `[(pid_type, pid_value(string)]` :/ 

where for the authors we do 
https://github.com/inspirehep/inspirehep/pull/3604/files#diff-f616257fa08231f84a2654e38bef46c7648caae30e21598a86c13f8f9f49b361R11

which returns correctly the control number as int